### DISCUSSION TITLE
请问开启clash情况下如何使用tcp进行dns解析？

### DISCUSSION THREAD

**Discussion by Bais-Huang** [vernesong/OpenClash#3739] [General]
当开启了clash后，我观察到我的dns forwarding被修改到了本地的7874端口，但是这个端口只启用了udp。
![image](https://github.com/vernesong/OpenClash/assets/25655255/e8c63302-9221-41c6-8b9d-436dd3c4810b)
可能是这个原因，导致了我通过tcp来进行dns解析时，会直接返回失败。
```
root@OpenWrt:/tmp# netstat -nap|grep 7874
udp        0      0 :::7874                 :::*                                6420/clash

```

我几乎可以确认是这个原因导致我的dns失败了，因为当我关闭clash时dns可以正常使用。

请问改如何配置clash可以让clash同时接受tcp的dns请求？

**Comment by ghost** [comment]
想请教一下，为何要用tcp传dns？
如果为了安全，DOH不是更好吗？

**Reply by Bais-Huang** [reply]
我使用的iCloudpd他在脚本里面写死了使用tcp做lookup。后来我直接把他的脚本改了就好了。
